cmd/sidecars/network-passt-binding/domain/configurator_test.go

```diff
@@ -347,6 +349,45 @@ var _ = Describe("pod network configurator", func() {
 
 			Expect(testMutator.Mutate(mutatedDomSpec)).To(Equal(mutatedDomSpec))
 		})
+	})
+	Context("should define memoryBacking for vhost-user", func() {
+		var testMutator *domain.PasstNetworkConfigurator
+		BeforeEach(func() {
+			networks := []vmschema.Network{*vmschema.DefaultPodNetwork()}
+			ifaces := []vmschema.Interface{{Name: "default", Binding: &vmschema.PluginBinding{Name: "passt"}}}
+
+			var err error
+			testMutator, err = domain.NewPasstNetworkConfigurator(ifaces, networks, domain.NetworkConfiguratorOptions{}, &netLinkStub{})
+			Expect(err).ToNot(HaveOccurred())
+		})
+
+		It("should set shared memfd when clean domain", func() {
+			mutatedDomSpec, err := testMutator.Mutate(&domainschema.DomainSpec{})
+			Expect(err).ToNot(HaveOccurred())
+			Expect(mutatedDomSpec.MemoryBacking.Access.Mode).To(Equal("shared"))
+			Expect(mutatedDomSpec.MemoryBacking.Source.Type).To(Equal("memfd"))
+		})
 
+		It("should fail when private memory is predefined", func() {
+			domainWithPrivateMem := &domainschema.DomainSpec{
+				MemoryBacking: &domainschema.MemoryBacking{
+					Access: &domainschema.MemoryBackingAccess{Mode: "private"},
+				},
+			}
+			_, err := testMutator.Mutate(domainWithPrivateMem)
+			Expect(err).To(HaveOccurred())
+		})
+
+		It("should use other configs of backing memory as long as they are shared", func() {
+			domainWithOtherSharedMem := &domainschema.DomainSpec{
+				MemoryBacking: &domainschema.MemoryBacking{
+					Access: &domainschema.MemoryBackingAccess{Mode: "shared"},
+					Source: &domainschema.MemoryBackingSource{Type: "file"},
+				},
+			}
+			mutatedDomSpec, err := testMutator.Mutate(domainWithOtherSharedMem)
+			Expect(err).NotTo(HaveOccurred())
+			Expect(mutatedDomSpec.MemoryBacking.Access.Mode).To(Equal("shared"))
```

Comment:
Please consider asserting that we haven't changed the whole `MemoryBacking` struct by asserting that it is equal to the initial state.